libshaderc/include/shaderc.h

```diff
@@ -113,6 +113,13 @@ void shaderc_compile_options_release(shaderc_compile_options_t options);
 void shaderc_compile_options_add_macro_definition(
     shaderc_compile_options_t options, const char* name, const char* value);
 
+// Set the compiler to only do preprocessing. When both preprocessing only mode
+// and disassembly mode are turned on, the compiler will respect preprocessing
+// only
```

Comment:
And yes, it's true the preprocessing-only mode overrides all other compilation modes (disassembly, and the default mode of compilation to SPIR-V binary).